src/Native/System.Native/pal_networking.cpp

```diff
@@ -2257,7 +2275,7 @@ static Error CloseSocketEventPortInner(int32_t port)
     return err == 0 ? PAL_SUCCESS : ConvertErrorPlatformToPal(errno);
 }
 
-static Error TryChangeSocketEventRegistrationInner(int32_t port, int32_t socket, SocketEvents currentEvents, SocketEvents newEvents, uintptr_t data)
+static Error TryChangeSocketEventRegistrationInner(int32_t port, int32_t socket, SocketEvents currentEvents, SocketEvents newEvents, int64_t data)
```

Comment:
What do you need the `#if/#else` guard for? It's not clear to me why any change would be necessary here.